### PR TITLE
Reimplemented Eon Hub

### PR DESCRIPTION
**Pull request by myersn024**
Reimplemented to use ContinuousRuleModifiyingEffectImpl in place of ReplacementEffectImpl. Tested and everything is working as intended.

### CODE REVIEW

File: Mage.Sets/src/mage/sets/fifthdawn/EonHub.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public EonHub copy() {
     
 }
 
-class SkipUpkeepStepEffect extends ReplacementEffectImpl {
+class SkipUpkeepStepEffect extends ContinuousRuleModifiyingEffectImpl {
     
     public SkipUpkeepStepEffect() {
         super(Duration.WhileOnBattlefield, Outcome.Neutral);
```
Comment (LevelX2) [inline Mage.Sets/src/mage/sets/fifthdawn/EonHub.java]: Check the now (now defaulted) log parameters. You probably won't have everytime a message to the player for this effect.

File: Mage.Sets/src/mage/sets/fifthdawn/EonHub.java [hunk 2]
```diff
@@ -57,24 +57,7 @@ public SkipUpkeepStepEffect copy() {
...
-    
-    @Override
-    public boolean checksEventType(GameEvent event, Game game) {
-        return event.getType() == GameEvent.EventType.UPKEEP_STEP;
-    }
-    
-    @Override
     public boolean applies(GameEvent event, Ability source, Game game) {
-        Player controller = game.getPlayer(source.getControllerId());
-        return controller != null && controller.getInRange().contains(event.getPlayerId());
```
Comment (LevelX2) [inline Mage.Sets/src/mage/sets/fifthdawn/EonHub.java]: So here simply return true if checksEventType is implemented;

**Comment by myersn024** [comment]
Made some changes based off your recommendations. I'm still not sure that I've got it exactly right in regards to the code implementation, but the card does function as intended.

**Comment by jeffwadsworth** [comment]
Good work.  By the way, have you tried generating the card code base via the “gen-card.pl” technique?

Check the bottom of this page for the Perl section.

https://github.com/magefree/mage/wiki/Developer-Getting-Started

**Comment by myersn024** [comment]
Jeff, I'll definitely look at the gen-card.pl script. Since I'm still really new to this project, what would be the best way to implement Cloud Key? I looked at ChooseModeEffect, but that seems to be only used in the siege cards and doesn't seem like it's the right way. Angelic Skirmisher has a similar kind of choose condition, but I want to start down the right path instead of having to re-implement after the fact.

**Pull request merged**